### PR TITLE
tests(pkg): beef up unit tests

### LINKED ISSUE #436: bring test coverage up to at least 70%
Here's where we stand today:

```
?       github.com/deis/builder/pkg     [no test files]
ok      github.com/deis/builder/pkg/cleaner     1.015s  coverage: 62.2% of statements
ok      github.com/deis/builder/pkg/conf        1.024s  coverage: 45.2% of statements
?       github.com/deis/builder/pkg/controller  [no test files]
ok      github.com/deis/builder/pkg/git 1.015s  coverage: 15.4% of statements
ok      github.com/deis/builder/pkg/gitreceive  1.021s  coverage: 37.3% of statements
ok      github.com/deis/builder/pkg/healthsrv   1.018s  coverage: 79.7% of statements
?       github.com/deis/builder/pkg/k8s [no test files]
ok      github.com/deis/builder/pkg/sshd        3.231s  coverage: 66.5% of statements
ok      github.com/deis/builder/pkg/storage     1.028s  coverage: 30.2% of statements
?       github.com/deis/builder/pkg/sys [no test files]
?       github.com/deis/builder [no test files]
```

There are a lot of packages that could see improvement through unit tests.

TODO:
 - [ ] bring total coverage up to >70%
 - [x] enable codecov on PRs

### PR DESCRIPTION
**Pull request by bacongobbler**
Using this PR along with codecov to display how much of the codebase is covered as I go along.

closes #436

### CODE REVIEW

**Comment by bacongobbler** [comment]
Something seems up with our test pipeline: https://travis-ci.org/deis/builder/builds/164114187#L292

**Comment by bacongobbler** [comment]
relies on https://github.com/deis/pkg/pull/27 being merged

**Comment by bacongobbler** [comment]
FYI codecov is reporting old news. Waiting on the jobs to go green first :)

**Comment by bacongobbler** [comment]
Here's where we stand in comparison to #436:

```
?   	github.com/deis/builder/pkg	[no test files]
ok  	github.com/deis/builder/pkg/cleaner	0.011s	coverage: 62.2% of statements
ok  	github.com/deis/builder/pkg/conf	0.004s	coverage: 89.3% of statements
ok  	github.com/deis/builder/pkg/controller	0.006s	coverage: 100.0% of statements
ok  	github.com/deis/builder/pkg/git	0.004s	coverage: 15.4% of statements
ok  	github.com/deis/builder/pkg/gitreceive	0.019s	coverage: 50.1% of statements
ok  	github.com/deis/builder/pkg/healthsrv	0.012s	coverage: 79.7% of statements
?   	github.com/deis/builder/pkg/k8s	[no test files]
ok  	github.com/deis/builder/pkg/sshd	1.868s	coverage: 68.5% of statements
ok  	github.com/deis/builder/pkg/storage	0.006s	coverage: 66.7% of statements
ok  	github.com/deis/builder/pkg/sys	0.003s	coverage: 100.0% of statements
?   	github.com/deis/builder	[no test files]
```

File: pkg/gitreceive/build.go [hunk 1]
```diff
@@ -62,7 +61,7 @@ func build(
 
 	slugBuilderImagePullPolicy, err := k8s.PullPolicyFromString(conf.SlugBuilderImagePullPolicy)
 	if err != nil {
-		return nil
+		return err
```
Comment (bacongobbler) [inline pkg/gitreceive/build.go]: Also noting that this effort caught an actual bug! :tada:

**Comment by bacongobbler** [comment]
ping @sgoings, do you think the existing test coverage is good enough to satisfy the current target or would you like to see a larger refactor of the codebase?

I've been trying to limit myself to just writing tests against what exists today rather than refactor legacy code and unintentionally break something. I don't think we're going to see any more gains in builder unless we start mocking the controller, kubernetes and s3 interactions, which means we'd be missing the original intent of tacking on unit tests to the existing infrastructure.

**Comment by mboersma** [comment]
> I've been trying to limit myself to just writing tests against what exists today rather than refactor legacy code

This is awesome IMHO, and it makes the coverage much better. We should revisit in the future to see if refactoring can allow more unit tests, but I think this is good enough to claim victory for this current dev cycle.

Although in the interests of full disclosure, your easter egg went right over my head. 😃

**Comment by bacongobbler** [comment]
> Although in the interests of full disclosure, your easter egg went right over my head. :smiley:

https://github.com/deis/deis/commit/0462cef5812ce31fe12f25596ff68dc614c708af

**Comment by mboersma** [comment]
> It's the sha of the very first commit to deis/deis

"Chef-based application fabric." Lol! Good times.

**Pull request merged**